packages/salesforcedx-vscode-apex/src/commands/metadataOrchestrator.ts

```diff
@@ -210,4 +217,70 @@ export class MetadataOrchestrator {
         return 'string';
     }
   };
+
+  public eligibilityDelegate = async (
+    requests: ApexOASEligiblePayload,
+    maxRetries: number = 3
+  ): Promise<ApexClassOASEligibleResponses> => {
+    const telemetryService = await getTelemetryService();
+    let response = {};
+    let attempt = 0;
+    const languageClient = languageClientUtils.getClientInstance();
+    if (languageClient) {
+      while (attempt < maxRetries) {
+        try {
+          attempt++;
+          response = await languageClient?.sendRequest('apexoas/isEligible', requests);
+          telemetryService.sendEventData('isEligibleResponseSucceeded', {
+            classNumbers: requests.payload.length.toString()
+          });
+          break;
+        } catch (error) {
+          telemetryService.sendException('isEligibleResponseFailed', `${error} after trying ${attempt} times.`);
+
+          if (attempt >= maxRetries) {
+            // fallback TBD after we understand it better
+            throw new Error(nls.localize('cannot_get_apexoaseligibility_response'));
+          }
+        }
+      }
+    }
+    return response as ApexClassOASEligibleResponses;
+  };
+
+  public validateEligibility = async (
+    sourceUri: vscode.Uri | vscode.Uri[],
+    isMethodSelected: boolean = false
+  ): Promise<ApexClassOASEligibleResponses> => {
+    const telemetryService = await getTelemetryService();
+    if (Array.isArray(sourceUri)) {
+      throw new Error('We do not consider list of src files now');
+    }
+    let cursorPosition;
+    if (isMethodSelected) {
+      const editor = vscode.window.activeTextEditor;
+      if (editor && this.getFileExtension(editor.document) === 'cls') {
+        cursorPosition = editor.selection.active;
+      } else {
+        telemetryService.sendException('activeTextEditorNotApex', nls.localize('active_text_editor_not_apex'));
+        throw new Error(nls.localize('invalid_active_text_editor'));
+      }
+    }
+    // generate the payload
+    const request: ApexClassOASEligibleRequest = {
+      resourceUri: sourceUri.path,
+      includeAllMethods: !isMethodSelected,
+      includeAllProperties: !isMethodSelected,
+      positions: cursorPosition ? [cursorPosition] : null,
+      methodNames: [],
+      propertyNames: []
+    };
+    const responses = await this.eligibilityDelegate({ payload: [request] });
+    return responses;
+  };
+
+  private getFileExtension(document: vscode.TextDocument): string {
```

Comment:
@mingxuanzhangsfdx this function is not needed.
It can be replaced with endsWith function call.

@CristiCanizales Why don't we condition the context and the this class commands using when property, then the supporting code can only run when active editor is a cls file.

Reply:
Ohh. I just realized the when clause is for right clicking in file explorer and editor but it's missing for command palette. Can you update that in this PR @mingxuanzhangsfdx ? It's line 261 in package.json